### ISSUE TITLE
Please add JID to Jinja Context

### ISSUE THREAD

**Issue by grichmond-salt** [saltstack/salt#49890]
### Description of Issue/Question
The job id is not presently available via Jinja during state runs.

Note for starting point from @gtmanfred:
  https://github.com/saltstack/salt/blob/2018.3/salt/modules/state.py#L1053

### Setup
N/A

### Steps to Reproduce Issue
N/A

### Versions Report
New request.

**Comment by gtmanfred** [comment]
Yeah, we should reasonably be able to pass the `__pub_jid` to the salt.utils.template.compile_template function so that the jid can be available in state files at the very least, then it can be passed to other templates using context.

**Comment by eliasp** [comment]
Isn't the JobID only available for non-local calls? So a `salt-call --local state.apply foo` would fail to render, as there's no JobID to be added to the Jinja context? So this would also effectively prevent local development of states outside a full-blown Minion/Master setup?

**Comment by gtmanfred** [comment]
the __pub_jid is the jid assigned from the master, so if you run --local, aka masterless, it will never hit the publish bus, and won't get the pub_jid.

**Comment by gtmanfred** [comment]
@saltstack/team-triage

**Comment by jtraub91** [comment]
This feature would be very beneficial for a use case I'm working with. Consider a case where we log into a device and create a session. We could make the session name include the job id so that we have an easier time cross referencing the execution on the remote device with the job results, located on the minion / master

**Comment by kajigga** [comment]
I am working on a solution where we need like to correlate logs generated by salt and logs created by processes that are triggered by salt (perhaps via cmd.run or something like that). 

It would be relatively easy to match up log entries if we could capture the `jid` and pass it to these sub processes where it could be included it in those generated logs.

**Comment by dkfsalt** [comment]
Having access to the JID would be useful for a use case of mine also

**Comment by sagetherage** [comment]
@waynew another good one for the Test Clinic, assigning you only in attempt to help write tests. We will still need tests written before we can commit to a release cycle for this issue and PR, but thank you all for your contributions!

**Comment by simmel** [comment]
`__orchestration_jid__` would also be nice to have
